src/plugins/booleans/and_operation.cpp

```diff
@@ -117,27 +118,27 @@ namespace phylanx { namespace execution_tree { namespace primitives
         }
 
         primitive_argument_type operator()(ir::node_data<double>&& lhs,
-            std::int64_t&& rhs) const
+            ir::node_data<std::int64_t>&& rhs) const
         {
             if (lhs.num_dimensions() != 0)
             {
                 return and_.and_all(
-                    std::move(lhs), ir::node_data<double>{rhs !=0 ? 1.0 : 0.0});
+                    std::move(lhs), ir::node_data<double>{rhs[0] !=0 ? 1.0 : 0.0});
```

Comment:
This code needs to be changed now as the comparison sematics have changed. Before we compared a `node_data<double>` with a single value, now we compare against a `node_data<int>`. Possible scenarios are now `Matrix<double> && Matrix<int>` etc. The same is probably true for all boolean operations...